base.rkt

```diff
@@ -212,27 +212,39 @@
       (bytes-set! pixels (+ 2 offset) green)
       (bytes-set! pixels (+ 3 offset) blue))))
 
-(define (flif->list dec-ptr)
+(define/contract (flif->list image)
+  (-> (or/c path-string? bytes?) (listof (is-a?/c bitmap%)))
+  ; create the decoder pointer
+  (define dec-ptr (flif-create-decoder))
+  ; decode either the file from the path or by its bytes
+  (if (path-string? image)
+      (flif-decoder-decode-file! dec-ptr image)
+      (flif-decoder-decode-memory! dec-ptr image))
   ; only look at the first frame if we want a static image
   (define num (if (want-animation?) (flif-decoder-num-images dec-ptr) 1))
-  (define dimensions (flif-dimensions (image-path)))
-  (for/list ([i (in-range num)])
-    (define image (flif-decoder-get-image dec-ptr i))
-    (define width (first dimensions))
-    (define height (second dimensions))
-    ; make sure to decode with the proper depth
-    (define reader (if (= (flif-image-get-depth image) 8)
-                       flif-image-read-rgba8
-                       flif-image-read-rgba16))
-    (define pixels (reader image width height))
-    (rgba->argb! pixels)
-    (define bitmap (make-object bitmap% width height #f #t))
-    (send bitmap set-argb-pixels 0 0 width height pixels)
-    bitmap))
+  ;(define width (flif-image-get-width image))
```

Comment:
Yeah, kill these lines. Could get confusing when casually perusing the code with these values "defined" in two places right by each other.